src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapper.tsx

```diff
@@ -0,0 +1,158 @@
+        return this.props.store.alterationEnrichmentRowData.filter(gene => {
+            return this.selectedGenes.includes(gene.hugoGeneSymbol);
+        });
+    }
+
+    public render() {
+        if (!this.PathwayMapperComponent) {
+            return null;
+        }
+
+        return (
+            <div>
+                <div className="alert alert-info">
+                    Ranking is based on top 10 genes having highest frequency by
+                    default. Genes of interest can be changed.
+                    <DefaultTooltip
+                        trigger={['click']}
+                        destroyTooltipOnHide={true}
+                        visible={this.isGeneSelectionPopupVisible}
+                        onVisibleChange={visible => {
+                            this.isGeneSelectionPopupVisible = !this
+                                .isGeneSelectionPopupVisible;
+                        }}
+                        overlay={
+                            <GenesSelection
+                                options={[]}
+                                selectedOption={{
+                                    label: GeneOptionLabel.USER_DEFINED_OPTION,
+                                    value: this.selectedGenes.join('\n'),
+                                }}
+                                onSelectedGenesChange={(
+                                    value,
+                                    genes,
+                                    label
+                                ) => {
+                                    this._genesOfInterest = genes.map(
+                                        g => g.gene
+                                    );
+                                    this.isGeneSelectionPopupVisible = false;
+                                }}
+                                defaultNumberOfGenes={10}
+                            />
+                        }
+                        placement="bottomLeft"
+                    >
+                        <button
+                            data-test="selectGenes"
+                            className="btn btn-default btn-xs"
+                            style={{ marginLeft: 5 }}
+                        >
+                            Select genes
+                        </button>
+                    </DefaultTooltip>
+                </div>
+                <div className="pathwayMapper">
+                    <div
+                        data-test="pathwayMapperTabDiv"
+                        className="cBioMode"
+                        style={{ width: '99%' }}
+                    >
+                        <Row>
+                            {/*
+                              // @ts-ignore */}
+                            <this.PathwayMapperComponent
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={this.genesOfInterest as any}
```

Comment:
why do we need the "any"?  Seems like types should be valid.

Reply:
For the PathwayMapper part, data are passed as "any". There are 3 different PM views now: ResultsView, GroupComparisonView, PatientView. Data format of genomic info is different for these 3 PM views. So, it is passed as "any". Patient and Result view have been already done before I started to my project and it was implemented in this way. So, I have continued in this way.